src/main/java/Assignment_3/FamilyTreeService.java

```diff
@@ -0,0 +1,153 @@
+package Assignment_3;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+
+
+public class FamilyTreeService {
+
+  private final HashMap<Integer, HashSet<Integer>> Edge;
+  private final HashMap<Integer, HashSet<Integer>> ReverseEdge;
+  private final HashMap<Integer, FamilyTreeModel> idToObject;
+
+  public FamilyTreeService() {
+    Edge = new HashMap<>();
+    ReverseEdge = new HashMap<>();
+    idToObject = new HashMap<>();
+  }
+
+  // validating Node Exist or not
+  void validateEdgeNodeId(int NodeId) {
```

Comment:
should move all validations into util class (create separate class for that, so every service can use the same to validate)